lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/LocalRuntimeDriverConfigurationGenerator.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.AvroConfigurationSerializer;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class LocalRuntimeDriverConfigurationGenerator {
+  private static final Logger LOG = Logger.getLogger(LocalRuntimeDriverConfigurationGenerator.class.getName());
+  private final REEFFileNames fileNames;
+  private final DriverConfigurationProvider driverConfigurationProvider;
+  private final Set<ConfigurationProvider> configurationProviders;
+  private final AvroConfigurationSerializer configurationSerializer;
+
+  @Inject
+  LocalRuntimeDriverConfigurationGenerator(final AvroConfigurationSerializer configurationSerializer,
+                                           final REEFFileNames fileNames,
+                                           final DriverConfigurationProvider driverConfigurationProvider,
+                                           @Parameter(DriverConfigurationProviders.class)
+                                           final Set<ConfigurationProvider> configurationProviders){
+    this.fileNames = fileNames;
+    this.driverConfigurationProvider = driverConfigurationProvider;
+    this.configurationProviders = configurationProviders;
+    this.configurationSerializer = configurationSerializer;
+  }
+
+  public Configuration writeConfiguration(final File jobFolder,
+                                          final String jobId,
+                                          final String clientRemoteId) throws IOException {
+    final File driverFolder = new File(jobFolder, PreparedDriverFolderLauncher.DRIVER_FOLDER_NAME);
+
+    final Configuration driverConfiguration1 = driverConfigurationProvider
+        .getDriverConfiguration(jobFolder, clientRemoteId,
+        jobId, Constants.DRIVER_CONFIGURATION_WITH_HTTP_AND_NAMESERVER);
+    final ConfigurationBuilder configurationBuilder = Tang.Factory.getTang().newConfigurationBuilder();
+    for (final ConfigurationProvider configurationProvider : this.configurationProviders) {
+      configurationBuilder.addConfiguration(configurationProvider.getConfiguration());
+    }
+    final Configuration providedConfigurations = configurationBuilder.build();
+    final Configuration driverConfiguration = Configurations.merge(
+        driverConfiguration1,
+        Tang.Factory.getTang()
+          .newConfigurationBuilder()
+          .bindNamedParameter(JobSubmissionDirectory.class, driverFolder.toString())
+          .build(),
+        providedConfigurations);
+    final File driverConfigurationFile = new File(driverFolder, fileNames.getDriverConfigurationPath());
+    configurationSerializer.toFile(driverConfiguration, driverConfigurationFile);
+    return driverConfiguration;
+  }
+
+  public static void main(final String[] args) throws InjectionException, IOException {
```

Comment:
The documentation for command line arguments was duplicated and already out of sync. I am consolidating them in one place and linking the others to that one place.

The .NET side call is going to be a subsequent PR I will add that to the documentation then.